server/src/main/java/com/hedera/block/server/verification/hasher/CommonUtils.java

```diff
@@ -0,0 +1,181 @@
+        }
+    }
+
+    /**
+     * Hashes the given left and right hashes.
+     * @param leftHash the left hash
+     * @param rightHash the right hash
+     * @return the combined hash
+     */
+    public static Bytes combine(@NonNull final Bytes leftHash, @NonNull final Bytes rightHash) {
+        return Bytes.wrap(combine(leftHash.toByteArray(), rightHash.toByteArray()));
+    }
+
+    /**
+     * Hashes the given left and right hashes.
+     * @param leftHash the left hash
+     * @param rightHash the right hash
+     * @return the combined hash
+     */
+    public static byte[] combine(final byte[] leftHash, final byte[] rightHash) {
+        try {
+            final var digest = MessageDigest.getInstance(DigestType.SHA_384.algorithmName());
+            digest.update(leftHash);
+            digest.update(rightHash);
+            return digest.digest();
+        } catch (final NoSuchAlgorithmException fatal) {
+            throw new IllegalStateException(fatal);
+        }
+    }
+
+    /**
+     * Returns the Hashes (input and output) of a list of block items.
+     * @param blockItems the block items
+     * @return the Hashes of the block items
+     */
+    public static Hashes getBlockHashes(List<BlockItemUnparsed> blockItems) {
+        int numInputs = 0;
+        int numOutputs = 0;
+        int itemSize = blockItems.size();
+        for (int i = 0; i < itemSize; i++) {
+            final BlockItemUnparsed item = blockItems.get(i);
+            final BlockItemUnparsed.ItemOneOfType kind = item.item().kind();
+            switch (kind) {
+                case EVENT_HEADER, EVENT_TRANSACTION -> numInputs++;
+                case TRANSACTION_RESULT, TRANSACTION_OUTPUT, STATE_CHANGES -> numOutputs++;
+            }
+        }
+
+        final var inputHashes = ByteBuffer.allocate(HASH_SIZE * numInputs);
+        final var outputHashes = ByteBuffer.allocate(HASH_SIZE * numOutputs);
+        final var digest = sha384DigestOrThrow();
+        for (int i = 0; i < itemSize; i++) {
+            final BlockItemUnparsed item = blockItems.get(i);
+            final BlockItemUnparsed.ItemOneOfType kind = item.item().kind();
+            switch (kind) {
+                case EVENT_HEADER, EVENT_TRANSACTION -> inputHashes.put(
+                        digest.digest(BlockItemUnparsed.PROTOBUF.toBytes(item).toByteArray()));
+                case TRANSACTION_RESULT, TRANSACTION_OUTPUT, STATE_CHANGES -> outputHashes.put(
+                        digest.digest(BlockItemUnparsed.PROTOBUF.toBytes(item).toByteArray()));
+            }
+        }
+
+        inputHashes.flip();
+        outputHashes.flip();
+
+        return new Hashes(inputHashes, outputHashes);
+    }
+
+    /**
+     * returns the ByteBuffer of the hash of the given block item.
+     * @param blockItemUnparsed the block item
+     * @return the ByteBuffer of the hash of the given block item
+     */
+    public static ByteBuffer getBlockItemHash(BlockItemUnparsed blockItemUnparsed) {
```

Comment:
could add the non-null annotation to the parameter for the static code analysis

Reply:
✅ Done!